### DISCUSSION TITLE
No button assignement possible

### DISCUSSION THREAD

**Discussion by FG2021** [xournalpp/xournalpp#3657] [Q&A] [answered]
I have a One by Wacom which is working just fine on my Ubuntu distribution with my laptop. Out of the box, plugging in the device and working with xournal++ is a charm, although it's version 0.5.something if I remember correctly.
Now I'm sitting in front of my Win10 desktop PC, the Wacom driver is installed and I have some issues.

First it was a hassle to get the buttons to work as intended (pressure sensitive tip, button 1 and 2). Buttons 1 and 2 were reversed in the settings menu. But I got everything to work as intended. The hover function of the graphics tablet is enabled. The buttons are set to "click" in the Wacom desktop center. However, there was one annoying issue. Everytime the pointer left the active page area or I used a keyboard shortcut (like Ctrl+S) the current stylus option was drawing where ever the cursor was without touching the tablet or pressing buttons. The resulting strokes were only "virtual" though. When actually pressing a button or using the pen tip, they would disappear and only the real input would be registered. Leaving the page area more than once though would make the previous "virtual" pen strokes "real". 

That is the reason why I tried the hotfix version. With this version however I cannot assign any pen style to the buttons. The pen has two buttons and I want button 1 to use the current tool, so I can draw with the hover function. The second button should switch to eraser. The stylus itself has no eraser on the other end. But that doesn't work at all. No matter what I set up in input system, even deactivating everything or setting whatever pen style to the buttons, all three inputs (tip and the two buttons) use the current tool.

![Input settings](https://user-images.githubusercontent.com/96004863/145710826-f5ffceca-b53e-419d-9bb4-695d58936029.PNG)
![Pen settings](https://user-images.githubusercontent.com/96004863/145710828-e90dee90-b602-4a8a-bc3e-59e72207efcf.PNG)
Am I doing anything wrong here, apart from using Windows that is?

**Comment by rolandlo** [comment, marked as answer]
For Wacom tablet users on Windows it may be better to either stick with version 1.1.0 (not the hotifx one) or to follow the advice here: https://github.com/xournalpp/xournalpp/issues/3218#issuecomment-891173434 setting the environment variable ` GDK_WIN32_TABLET_INPUT_API` to `WinTab`.
The hotfix version uses a newer pointer input API (than WinTab) that works better on almost all devices except Wacom Tablets.

Some issues with "virtual" strokes when crossing the page borders are known, see https://github.com/xournalpp/xournalpp/issues/2762, although yet unresolved.

**Reply by FG2021** [reply]
The only thing I observed is that these virtual lines, or whatever tool you are using, only seem to appear when the last action with the pen was one where you use the hover functionality. Using my 2nd button with the eraser and hovering to erase something will cause virtual lines when crossing page borders etc. When the last action was coming from the tip you don't get these virtual lines.
That does only apply to the page region though. Using the pen tip to select a tool from the toolbar does not count. 

The most annoying part about this is the use of the vertical space tool. If the last action was hover and you select the vertical space it will automatically start with this tools action upon entering the page area. Simply clicking into the page will delete its entire content and there is no way to get it back for me. It's just gone. When this happens one has to leave the page at the top and click on the space between pages.

I just now realize that I left my "place holder" title for this topic.
